src/v/redpanda/admin_server.cc

```diff
@@ -2324,6 +2324,29 @@ admin_server::stop_broker_maintenance_handler(
           "progress?)");
     }
     model::node_id id = parse_broker_id(*req);
+
+    // RPK will check that the broker is actively draining before
+    // executing the disable, so we do the same here. For details, see
+    // src/go/rpk/pkg/cli/cluster/maintenance/disable.go
+    auto maybe_drain_status = co_await get_node_drain_status(
+      _controller, id, model::time_from_now(5s));
+    if (!maybe_drain_status) {
+        vlog(
+          logger.info,
+          "Maintenance mode is already disabled for node {}, refreshing disk "
+          "health.",
+          id);
+
+        // Maintenance status is determined by the health cache on
+        // non-leader nodes. And maintenance disable may fail
+        // because the health cache was not updated yet. See this
+        // link for details
+        // https://github.com/redpanda-data/redpanda/issues/9284#issuecomment-1456927970
+        // Therefore, force update the health cache before we continue with
+        // maintenance disable
+        co_await refresh_disk_health_handler();
+    }
+
```

Comment:
I realize why this is failing in CI still, because control first goes through RPK and the checks in there are failing before anything is sent to Redpanda.

Reply:
Here's an idea: instead of refreshing disk health within the call to disable maintenance mode, we could check for not draining status in `get_local_maintenance` and refresh disk health before we return.

The consequences of this change are:
1. In this DT test, many methods call `get_local_maintenance` before the call to disable maintenance mode, so disk health **should** be up-to-date by then.
2. I imagine that users would query maintenance status at some point before enabling/disabling maintenance mode, so disk health **should** be up-to-date. Of course, there is no guarantee that a user queries maintenance status but how else can we auto-update disk health **and** bypass the checks in `rpk`? I am open to ideas.